### PR TITLE
Runtime warning

### PR DESCRIPTION
**Pull request by ahn1340**
Introduces new way of computing training loss for cross-validation.

### CODE REVIEW

**Comment by ahn1340** [comment]
@mfeurer, sorry for the really late notice. The PR is done but Travis still fails due to the xgboost errors. Other than that, I believe the code works fine. Please have a look at it and let me know.

**Comment by mfeurer** [comment]
This looks good, thanks a lot @ahn1340.

Sorry that it took me so long to check this, too.
